source/common/network/utility.cc

```diff
@@ -88,24 +88,68 @@ Api::IoCallUint64Result receiveMessage(uint64_t max_rx_datagram_size, Buffer::In
   return result;
 }
 
-} // namespace
-
-Address::InstanceConstSharedPtr Utility::parseInternetAddressNoThrow(const std::string& ip_address,
-                                                                     uint16_t port, bool v6only) {
+StatusOr<sockaddr_in> parseV4Address(const std::string& ip_address, uint16_t port) {
   sockaddr_in sa4;
   memset(&sa4, 0, sizeof(sa4));
-  if (inet_pton(AF_INET, ip_address.c_str(), &sa4.sin_addr) == 1) {
-    sa4.sin_family = AF_INET;
-    sa4.sin_port = htons(port);
-    return instanceOrNull(Address::InstanceFactory::createInstancePtr<Address::Ipv4Instance>(&sa4));
+  if (inet_pton(AF_INET, ip_address.c_str(), &sa4.sin_addr) != 1) {
+    return absl::FailedPreconditionError("failed parsing ipv4");
   }
+  sa4.sin_family = AF_INET;
+  sa4.sin_port = htons(port);
+  return sa4;
+}
+
+StatusOr<sockaddr_in6> parseV6Address(const std::string& ip_address, uint16_t port) {
   sockaddr_in6 sa6;
   memset(&sa6, 0, sizeof(sa6));
-  if (inet_pton(AF_INET6, ip_address.c_str(), &sa6.sin6_addr) == 1) {
+  const auto scope_pos = ip_address.rfind('%');
+  // Even though it would be nice to do any IPv6 parsing only with the getaddrinfo at the moment
+  // Windows parsing is slightly different in behavior than other platforms. For this reason we use
+  // the inet_pton for any parsing that does not contain the zone id.
```

Comment:
Please tag with TODO the places we will cleanup with https://github.com/envoyproxy/envoy/issues/23952.

Reply:
done